### ISSUE TITLE
[BUG] Memory leak

### ISSUE THREAD

**Issue by adilansari** [dlvhdr/gh-dash#491]
**Describe the bug**
I have seen it taking upto 8 GB of RAM when running overtime. I usually have dashboard always open in one of the tmux panes with auto refresh set to 2 minutes. After almost a day the memory footprint is 8 GB, at that point I just restart gh-dash.

**To Reproduce**
1. Run gh-dash for almost a day and see its RAM usage grow overtime.

> screenshot after 12 hours, its at 1.35 GB
<img width="589" alt="image" src="https://github.com/user-attachments/assets/c26360f6-adbb-4c30-aad7-741a2b003ddf" />

> after 3 days
<img width="673" alt="image" src="https://github.com/user-attachments/assets/72de8f07-9d1c-4c09-a9f0-8196ca3cd79f" />

**Expected behavior**
It is supposed to be stateless and not consume this much memory

**Desktop (please complete the following information):**
 - MacOS
 - Tmux

**Comment by dlvhdr** [comment]
Yes I can reproduce but not sure how to debug.. will try and do some reading

**Comment by dlvhdr** [comment]
attempting a fix in #495

**Comment by adilansari** [comment]
Thanks for addressing the bug. I got the latest release, will bake it for a couple days and update here if the issue still persists.

**Comment by adilansari** [comment]
<img width="578" alt="image" src="https://github.com/user-attachments/assets/1075946e-f8a1-473a-b8ea-e174d94f58d1" />

2 gigs after 2 days... Definitely better but still think its too much. Among the top 3 memory intensive processes on my machine.

**Comment by dlvhdr** [comment]
@adilansari, can you try with this latest version?

**Comment by adilansari** [comment]
It has been much better now. 4 days and memory usage is just 35 MB. Night and day, just curious what profiler/techniques did you use to uncover this? I'd like to learn

**Comment by dlvhdr** [comment]
You can follow in the go discord

https://discord.gg/golang

https://discord.com/channels/118456055842734083/1327697352336736257/1327697352336736257
